Use browserify for building the browser version

joseph-onsip opened the pull request:
This resolves https://github.com/pegjs/pegjs/issues/373 and,
since `browserify` produces a UMD bundle (due to `--standalone PEG`),
addresses the first part of https://github.com/pegjs/pegjs/issues/362,

> 1. Making PEG.js itself UMD module.

dmajda commented:
@joseph-onsip Nice patch, thanks! Could you please fix the small nit I pointed out in the [other comment](https://github.com/pegjs/pegjs/pull/379#commitcomment-13817973) and rebase (so the change is still one commit)? I’ll merge then.

I also compared browser build sizes with the current code and after applying your patch:

**Before**

```
282660 peg-0.9.0.js
122389 peg-0.9.0.min.js
``` 

**After**

```
256077 peg-0.9.0.js
123484 peg-0.9.0.min.js
```

The difference in case of `peg-0.9.0.js` is caused mostly by indentation whitespace (the current code indents the modules, Browserify does not). The difference in case of `peg-0.9.0.min.js` is Browserify’s overhead. Although I’m not much happy about it I think it is small enough to live with it.

joseph-onsip commented:
Sure thing, I've added `MAIN_FILE` and squashed all changes into 0d8c0458232339e2bc399c01140561d14e6a8058.

I also noticed the change in build sizes, and was a little irked by the increase in `peg-0.9.0.min.js`, but after realizing that it's also producing a UMD module (for #362), I figured it was worth it.

dmajda commented:
Thanks, merging now.
